semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/messaging/MessagingSemanticConventionUtils.java

```diff
@@ -34,7 +53,80 @@
   }
 
   public static boolean isRabbitMqBackend(Event event) {
+    if(SpanAttributeUtils.containsAttributeKey(event, MESSAGING_SYSTEM)) {
+      return RABBITMQ_SYSTEM_VALUE.equals(SpanAttributeUtils.getStringAttributeWithDefault(
+          event, MESSAGING_SYSTEM, StringUtils.EMPTY));
+    }
+
     return SpanAttributeUtils.containsAttributeKey(event, RawSpanConstants.getValue(RabbitMq.RABBIT_MQ_ROUTING_KEY))
         || SpanAttributeUtils.containsAttributeKey(event, OtelMessagingSemanticConventions.RABBITMQ_ROUTING_KEY.getValue());
   }
+
+  public static Optional<String> getKafkaBackendURI(Event event) {
+    if (!isKafkaBackend(event)) {
+      return Optional.empty();
+    }
+
+    if (SpanAttributeUtils.containsAttributeKey(event, MESSAGING_URL)) {
+      return Optional.of(SpanAttributeUtils.getStringAttribute(event, MESSAGING_URL));
+    }
+
+    if (DbSemanticConventionUtils.getBackendURIForOpenTracingFormat(event).isPresent()) {
+      return DbSemanticConventionUtils.getBackendURIForOpenTracingFormat(event);
+    }
+
+    return DbSemanticConventionUtils.getBackendURIForOtelFormat(event);
+  }
+
+  public static boolean isKafkaBackend(Event event) {
+    if(SpanAttributeUtils.containsAttributeKey(event, MESSAGING_SYSTEM)) {
+      return KAFKA_SYSTEM_VALUE.equals(SpanAttributeUtils.getStringAttributeWithDefault(
+          event, MESSAGING_SYSTEM, StringUtils.EMPTY));
+    }
+
+    if(SpanAttributeUtils.containsAttributeKey(event, PEER_SERVICE_NAME)) {
+      return KAFKA_SYSTEM_VALUE.equals(SpanAttributeUtils.getStringAttributeWithDefault(
+          event, PEER_SERVICE_NAME, StringUtils.EMPTY));
+    }
+
+    return false;
+  }
+
+  public static Optional<String> getSqsBackendURI(Event event) {
+    if (!isSqsBackend(event)) {
+      return Optional.empty();
+    }
+
+    if(SpanAttributeUtils.containsAttributeKey(event, MESSAGING_URL)) {
```

Comment:
Can we move this try/catch logic to a small function?
`getURIFromMessageUrl(String)`